### PR TITLE
Add functions for determining the maximum useful thread count

### PR DESCRIPTION
**Pull request by Xaeroxe**
This could be useful for reducing CPU overhead in Amethyst

### CODE REVIEW

File: src/dispatch/dispatcher.rs [hunk 1]
```diff
@@ -94,6 +94,18 @@ impl<'a, 'b> Dispatcher<'a, 'b> {
...
+    /// This method returns the largest amount of threads this dispatcher
+    /// can make use of.  Allocating anymore threads than this will use CPU
+    /// without conferring any benefit.  (Unless something is running in
+    /// the threadpool other than the dispatcher.)
+    #[cfg(feature = "parallel")]
+    pub fn max_threads(&self) -> usize {
+        self.stages
+            .iter()
+            .map(|s| s.max_threads())
+            .fold(0, |highest, value| highest.max(value))
```
Comment (torkleyy) [inline src/dispatch/dispatcher.rs]: Would rayon permit creating 0 threads?

**Comment by torkleyy** [comment]
I'm not sure I can recommend this as there are other things than the
dispatcher running jobs with the pool. E.g. asset loading

File: src/dispatch/dispatcher.rs (hunk 1 above)
Comment (Xaeroxe) [reply src/dispatch/dispatcher.rs]: Possibly not, but nonetheless the API report is accurate.  A dispatcher with nothing to do needs no threads.

**Comment by Xaeroxe** [comment]
@torkleyy right, which is why Amethyst would add 1 or 2 to the value reported by this API in order to accomodate that.

**Comment by Xaeroxe** [comment]
@torkleyy quick context for the motivation behind this: https://github.com/amethyst/amethyst/issues/780#issuecomment-396281153

**Comment by torkleyy** [comment]
I saw that, but I'm still not convinced it makes sense to create a pool with less threads than cores, and I don't think it's the solution to the high CPU usage.

**Pull request merged**